src/Adapters/AzureManagementAdapter.ts

```diff
@@ -567,4 +571,94 @@ export default class AzureManagementAdapter implements IAzureManagementAdapter {
             return new AzureResourcesData([newRoleAssignment]);
         }, 'azureManagement');
     }
+
+    async getTimeSeriesConnectionInformation(adtInstanceIdentifier: {
+        id?: string;
+        hostName?: string;
+    }) {
+        // either the resource object or host url of the ADT instance
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+
+        return await adapterMethodSandbox.safelyFetchData(async (token) => {
+            let adtInstance: IAzureResource;
+            if (adtInstanceIdentifier.hostName) {
+                const digitalTwinInstances = await this.getResourcesByPermissions(
+                    {
+                        getResourcesParams: {
+                            resourceType: AzureResourceTypes.DigitalTwinInstance
+                        },
+                        requiredAccessRoles: RequiredAccessRoleGroupForADTInstance
+                    }
+                );
+                const result = digitalTwinInstances.result.data;
+                adtInstance = result.find(
+                    (d) =>
+                        d.properties.hostName === adtInstanceIdentifier.hostName
+                );
+            } else if (adtInstanceIdentifier.id) {
+                const digitalTwinInstance = await this.getResourceById(
+                    adtInstanceIdentifier.id
+                );
+                adtInstance = digitalTwinInstance.result.data;
+            }
+
+            try {
+                const url = `https://management.azure.com${adtInstance.id}/timeSeriesDatabaseConnections`;
+                const timeSeriesDatabaseConnections: Array<IAzureTimeSeriesDatabaseConnection> = await this.fetchAllResources(
+                    adapterMethodSandbox,
+                    [],
+                    token,
+                    url,
+                    {
+                        apiVersion:
+                            AzureResourcesAPIVersions[
+                                'Microsoft.DigitalTwins/digitalTwinsInstances/timeSeriesDatabaseConnections'
+                            ]
+                    }
+                );
+
+                const connectionData = timeSeriesDatabaseConnections[0]; // TODO: for now get the first connection information
```

Comment:
do we need to assert that we got at least 1 connection back?

Reply:
good point, no since it is wrapped in try catch, if it fails it will fall into catch block which returns null